src/diffusers/loaders.py

```diff
@@ -1115,14 +1115,18 @@ def load_lora_weights(self, pretrained_model_name_or_path_or_dict: Union[str, Di
             low_cpu_mem_usage=low_cpu_mem_usage,
             _pipeline=self,
         )
-        self.load_lora_into_text_encoder(
-            state_dict,
-            network_alphas=network_alphas,
-            text_encoder=self.text_encoder,
-            lora_scale=self.lora_scale,
-            low_cpu_mem_usage=low_cpu_mem_usage,
-            _pipeline=self,
-        )
+
+        if hasattr(self, "text_encoder"):
+            self.load_lora_into_text_encoder(
+                state_dict,
+                network_alphas=network_alphas,
+                text_encoder=self.text_encoder,
+                lora_scale=self.lora_scale,
+                low_cpu_mem_usage=low_cpu_mem_usage,
+                _pipeline=self,
+            )
+        else:
+            logger.warn("No `text_encoder` present in this pipeline, skipping `load_lora_into_text_encoder`.")
```

Comment:
```suggestion
        if hasattr(self, "text_encoder"):
            self.load_lora_into_text_encoder(
                state_dict,
                network_alphas=network_alphas,
                text_encoder=self.text_encoder,
                lora_scale=self.lora_scale,
                low_cpu_mem_usage=low_cpu_mem_usage,
                _pipeline=self,
            )
        else:
            logger.warn("No `text_encoder` present in this pipeline, skipping `load_lora_into_text_encoder`.")
```

I think this method should fail if the model has no text_encoder - the method becomes too hacky otherwise

Reply:
Hey Patrick!

I was hesitant and I agree with you: but there _are_ diffusers (paint by example in this case, there might be others?) that are unet only or have an image encoder rather a text encoder; is there another way to load lora weights in such case?

I was also thinking of maybe doing a try except according to this: https://stackoverflow.com/a/610923/7953481 not sure if it would help.

Reply:
Hmm, I guess we could show in an example how the user could run the two functions
```
 state_dict, network_alphas = self.lora_state_dict(pretrained_model_name_or_path_or_dict, **kwargs)
        self.load_lora_into_unet(
            state_dict,
            network_alphas=network_alphas,
            unet=self.unet,
            low_cpu_mem_usage=low_cpu_mem_usage,
            _pipeline=self,
        )
```

invidually. Did you get some decent results with LoRA + paint by example?

Reply:
I did get some decent results with paint-by-example + LoRA 😆 

I was thinking two alternatives, first is like you suggested, to do `lora_state_dict` and then `load_lora_into_unet`; the other is to be similar like sdxl below, to only call `load_lora_into_text_encoder` when the LoRA itself has the text encoder parts in the state dict (which could prevent LoRA weights with text encoder to be used, in which case I was able to still get decent results)